pkg/securitypolicy/securitypolicyenforcer.go

```diff
@@ -32,26 +50,64 @@ type SecurityPolicyEnforcer interface {
 	EncodedSecurityPolicy() string
 }
 
-func NewSecurityPolicyEnforcer(state SecurityPolicyState, eOpts ...standardEnforcerOpt) (SecurityPolicyEnforcer, error) {
-	if state.SecurityPolicy.AllowAll {
-		if state.SecurityPolicy.Containers.Length > 0 || len(state.SecurityPolicy.Containers.Elements) > 0 {
-			return nil, ErrInvalidAllowAllPolicy
+// CreateAllowAllEnforcer creates and returns OpenDoorSecurityPolicyEnforcer instance.
+// Both AllowAll and Containers cannot be set at the same time.
+func CreateAllowAllEnforcer(state SecurityPolicyState, _, _ []oci.Mount) (SecurityPolicyEnforcer, error) {
+	policyContainers := state.SecurityPolicy.Containers
+	if !state.AllowAll || policyContainers.Length > 0 || len(policyContainers.Elements) > 0 {
+		return nil, ErrInvalidAllowAllPolicy
+	}
+	return &OpenDoorSecurityPolicyEnforcer{
+		encodedSecurityPolicy: state.EncodedSecurityPolicy.SecurityPolicy,
+	}, nil
+}
+
+// CreateStandardEnforcer creates and returns StandardSecurityPolicyEnforcer instance.
+// Make sure that the input JSON policy can be converted to internal representation
+// and that `criMounts` and `criPrivilegedMounts` can be injected before successful return.
+func CreateStandardEnforcer(
```

Comment:
do we want to make `CreateSecurityPolicyEnforcer` the main entrypoint here, and unexport these functions, (and `CreateEnforcerFunc`)?
That will likely be easier for use to maintain

Reply:
FYI this code has a lifespan that is measured in weeks.

Reply:
I think it makes sense. we're not using those outside of this package.